--- ./program/psiconv/psiconv.c-old	2020-09-18 21:17:20.026326840 -0400
+++ ./program/psiconv/psiconv.c	2020-09-18 21:18:23.944022095 -0400
@@ -44,6 +44,8 @@
 #include "psiconv.h"
 #include "gen.h"
 
+psiconv_list fileformat_list;
+
 static void print_help(void);
 static void print_version(void);
 static void strtoupper(char *str);
--- ./program/psiconv/psiconv.h-old	2020-09-18 21:17:11.887238307 -0400
+++ ./program/psiconv/psiconv.h	2020-09-18 21:17:54.612703049 -0400
@@ -52,7 +52,7 @@
   output_function *output;
 } *fileformat;
 
-psiconv_list fileformat_list; /* of struct psiconv_fileformat */
+extern psiconv_list fileformat_list; /* of struct psiconv_fileformat */
 
 
 #endif /* PSICONV_H */
